dotnet/src/SemanticKernel.Core/Extensions/SKFunctionExtensions.cs

```diff
@@ -20,19 +20,16 @@ public static class SKFunctionExtensions
     /// <param name="function">Function to execute</param>
     /// <param name="kernel">Kernel</param>
     /// <param name="variables">Input variables for the function</param>
-    /// <param name="plugins">Collection of plugins that this function can access</param>
     /// <param name="requestSettings">LLM completion settings (for semantic functions only)</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>The result of the function execution</returns>
     public static Task<FunctionResult> InvokeAsync(this KernelFunction function,
         Kernel kernel,
         ContextVariables? variables = null,
-        IReadOnlySKPluginCollection? plugins = null,
         AIRequestSettings? requestSettings = null,
         CancellationToken cancellationToken = default)
     {
-        var context = kernel.CreateNewContext(variables, plugins);
-        return function.InvokeAsync(kernel, context, requestSettings, cancellationToken);
+        return function.InvokeAsync(kernel, variables ?? new ContextVariables(), requestSettings, cancellationToken);
```

Comment:
Without the plugins I believe this extension can go away and the ContextVariables there should be like here optional.

Reply:
I'll do it in one of the PRs when replacing parameters of ContextVariables type by KernelFunctionArguments or the one afterwards.